apps/src/music/player/MusicLibrary.ts

```diff
@@ -423,32 +450,44 @@ export type SoundFolderType = 'sound' | 'kit' | 'instrument';
 
 export interface SoundFolder {
   name: string;
-  artist?: string;
   id: string;
-  type?: SoundFolderType;
   path: string;
-  imageSrc: string;
-  color?: string;
-  restricted?: boolean;
+  imageAttribution: ImageAttribution;
+  skipLocalization: boolean;
   sounds: SoundData[];
-  bpm?: number;
-  key?: Key;
-  imageAttribution?: ImageAttribution;
-  skipLocalization?: boolean;
 }
 
+export interface SoundFolderPack extends SoundFolder {
+  artist: string;
+  imageSrc: string;
+  color: string;
+  restricted: boolean;
+  bpm: number;
+  key: Key;
+  sounds: SoundDataPack[];
+}
+
+export interface SoundFolderInstrument extends SoundFolder {
+  type: SoundFolderType;
+  sounds: SoundDataInstrument[];
+}
+
+export interface SoundFolderKit extends SoundFolder {
+  type: SoundFolderType;
+  sounds: SoundDataKit[];
+}
+
+type SoundFolderAny = SoundFolderPack | SoundFolderInstrument | SoundFolderKit;
```

Comment:
nit: maybe rename the base interface something like `BaseSoundFolder` or `SoundFolderCommon` and keep this `SoundFolder`?